### PR TITLE
feat: DIPA-215 header data project

### PR DESCRIPTION
**Pull request by irenenguyen3095**
- DIPA-215: new attributes for project table
- DIPA-214: (backend) save data for project Feldberg
- DIPA-226: (backend) update project's data
- DIPA-211: (backend) project size

### CODE REVIEW

File: server/src/main/java/online/dipa/hub/convert/ProjectTemplateToProjectConverter.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                             .name(template.getName())
+                             .akz(template.getAKZ())
+                             .client(template.getClient())
+                             .department(template.getDepartment())
+                             .projectOwner(template.getProjectOwner());
+        
+        if (template.getProjectSize() != null) {
+            project.projectSize(Project.ProjectSizeEnum.fromValue(template.getProjectSize()));
+        }
+        return project;
```
Comment (Bock4Soft) [inline server/src/main/java/online/dipa/hub/convert/ProjectTemplateToProjectConverter.java:26]: return new Project().id(template.getId())
                             .name(template.getName())
                             .akz(template.getAKZ())
                             .client(template.getClient())
                             .department(template.getDepartment())
                             .projectOwner(template.getProjectOwner())
                             .projectSize(Project.ProjectSizeEnum.fromValue(template.getProjectSize()));

File: server/src/main/java/online/dipa/hub/services/ProjectService.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                        String.format("Project with id: %1$s not found.", projectId)));
+    }
+
+    
+    private void initializeProjects() {
+
+        projectRespository.findAll().stream().map(p -> conversionService.convert(p, Project.class))
+                .forEach(t -> {
+                    ProjectState sessionProject = findProjectState(t.getId());
+                    if (sessionProject.getProject() == null) {
```
Comment (Bock4Soft) [inline server/src/main/java/online/dipa/hub/services/ProjectService.java:48]: t.getId() kann potenziell ne NullPointerException werfen. Kann man damit verhindern:

.filter(Objects::nonNull)
                .forEach(t -> {

File: server/src/main/java/online/dipa/hub/convert/ProjectTemplateToTimelineConverter.java [hunk 3]
```diff
@@ -20,7 +20,7 @@
 import java.util.stream.Collectors;
 
 @Component
-public class ProjectToTimelineConverter implements Converter<ProjectEntity, Timeline> {
+public class ProjectTemplateToTimelineConverter implements Converter<ProjectEntity, Timeline> {
```
Comment (Bock4Soft) [inline server/src/main/java/online/dipa/hub/convert/ProjectTemplateToTimelineConverter.java:23]: Eher ProjectEntityToTimelineConverter, oder?

File: server/src/main/java/online/dipa/hub/convert/ProjectTemplateToProjectConverter.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.stereotype.Component;
+
+import online.dipa.hub.api.model.Project;
+
+import online.dipa.hub.persistence.entities.ProjectEntity;
+
+@Component
+public class ProjectTemplateToProjectConverter implements Converter<ProjectEntity, Project> {
```
Comment (Bock4Soft) [inline server/src/main/java/online/dipa/hub/convert/ProjectTemplateToProjectConverter.java:11]: Eher ProjectEntityToProjectConverter, oder?

**Review by Bock4Soft — CHANGES_REQUESTED** [review]
Ansonsten sieht es gut aus

**Review by irenenguyen3095 — COMMENTED** [review]
sieht gut aus!

**Pull request merged**